packages/core-components/src/layout/SignInPage/UserIdentity.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  IdentityApi,
+  ProfileInfo,
+  ProfileInfoApi,
+  BackstageUserIdentity,
+  BackstageIdentityApi,
+  SessionApi,
+} from '@backstage/core-plugin-api';
+
+export class UserIdentity implements IdentityApi {
+  static from(options: {
+    identity: BackstageUserIdentity;
+    authApi: ProfileInfoApi & BackstageIdentityApi & SessionApi;
+    /**
+     * Passing a profile synchronously allows the deprecated `getProfile` method to be
+     * called by consumers of the {@link IdentityApi}. If you do not have any consumers
+     * of that method than this is safe to leave out.
+     *
+     * @deprecated Only provide this if you have plugins that call the synchronous `getProfile` method, which is also deprecated.
+     */
+    profile?: ProfileInfo;
+  }) {
+    return new UserIdentity(options.identity, options.authApi, options.profile);
+  }
+
+  private constructor(
+    private readonly identity: BackstageUserIdentity,
+    private readonly authApi: ProfileInfoApi &
+      BackstageIdentityApi &
+      SessionApi,
+    private readonly profile?: ProfileInfo,
+  ) {}
+
+  getUserId(): string {
+    const ref = this.identity.userEntityRef;
+    const match = /^([^:/]+:)?([^:/]+\/)?([^:/]+)$/.exec(ref);
```

Comment:
should use `parseRefString` from `catalog-model` instead

Reply:
We opted for not adding that dependency

Reply:
This is also deprecated so gonna disappear long-term